src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -796,9 +796,14 @@ public TypeSpec build() {
 
       for (MethodSpec methodSpec : methodSpecs) {
         if (kind == Kind.INTERFACE) {
-          requireExactlyOneOf(methodSpec.modifiers, Modifier.ABSTRACT, Modifier.STATIC,
-              Modifier.DEFAULT);
           requireExactlyOneOf(methodSpec.modifiers, Modifier.PUBLIC, Modifier.PRIVATE);
+          if (methodSpec.modifiers.contains(Modifier.PRIVATE)) {
+            checkState(!methodSpec.hasModifier(Modifier.DEFAULT), "%s %s.%s cannot be private and default",
```

Comment:
Potentially also check that it's not `ABSTRACT`? I don't think that's valid either.

Though perhaps a check like that belongs higher up, since it applies to _all_ `private` methods, not just ones in interfaces.